enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,350 @@
+    // encrypt volumes.
+    // +optional
+    EncryptionKey *EncryptionKeyReference `json:"encryptionKey,omitempty"`
+}
+```
+
+#### Install Config
+
+The API extensions above correspond to types already in use in the Installer install config: 
+[AWS](https://github.com/openshift/installer/blob/master/pkg/types/aws/machinepool.go#L100), 
+[GCP](https://github.com/openshift/installer/blob/master/pkg/types/gcp/machinepools.go#L73), &
+[Azure](https://github.com/openshift/installer/blob/master/pkg/types/azure/disk.go#L36).
+
+### Implementation Details/Notes/Constraints
+
+The Installer will write the encryption keys to the cluster infrastructure object when users have defined
+encryption keys on the `defaultMachinePlatform` (`installconfig.platform.(aws|azure|gcp).defaultMachinePlatform`).
+
+1. User adds encryption keys to the `defaultMachinePlatform` in the install config.
+2. Installer checks if `defaultMachinePlatform` has encryption keys defined, sets them in cluster infrastructure object.
+3. Cluster infrastructure object is created during install.
+4. cluster-storage-operator/CSI driver monitors cluster infrastructure object and creates default storage class
+accordingly
+5. Should the API extension be more broad to encompass a general storage class, rather than specifically geared toward
+encryption?
+
+### Risks and Mitigations
+
+No risks foreseen. 
+
+### Drawbacks
+
+These tasks could be handled day-2 by a user. The largest drawback would be in terms of the added complexity required to
+implement this feature when that could be handled day 2 by a user. The complexity of this task is quite small and seems
+reasonable, so this drawback does not seem like a reason to block this enhancement.
+
+## Design Details
+
+### Open Questions 
+
+1. Should these API extensions go in both the spec & status as proposed?
+2. When should the installer write the keys to the cluster infrastructure object; particularly which fields in the
+install config correlate to "set default storage class"? Is the `defaultMachinePool`, as proposed the best fit? What
+about if control-plane/compute nodes only are set? What about a new field in the install config
+([see alternatives below](#alternatives)).
+3. Should e2e-tests be required for this feature?
+
+### Test Plan
+
+If desired, presubmit and/or periodic jobs could follow this pattern:
+1. In pre-steps, create or use existing keys to populate install config.
+2. Create cluster
+3. Add a step to ensure default storage class uses specified keys.
+
+### Graduation Criteria
+
+#### GA
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+### Upgrade / Downgrade Strategy
+
+After upgrade, users should be able to utilize the new API fields to have the cluster-storage-operator or CSI driver
```

Comment:
Does the CSO have it's own operator configuration? We had a similar feature recently proposed within the infrastructure object that we concluded would live better closer to the operator (default ingress type), so I'm wondering if this belongs in the infra or whether it belongs in the CSO operator resource

Reply:
Writing this config to a storage-specific resource sounds like a better approach than writing to the infrastructure object, but I have a couple of questions:
 
1. Can users edit this ClusterCSIDriver (day 2)? 
2. Will the storage operator properly reconcile a config laid down by the installer--i.e. the installer-created config would not be overwritten by the operator?

Reply:
1. Yes, users can edit ClusterCSIDriver after installation
2. We can make our operators properly reconcile ClusterCSIDriver, so the version created by user or installer is intact.

Reply:
I think currently our operator will only create `ClusterCSIDriver` object if it does not already exist by the looks of it.

Reply:
Ok so sounds like going straight to creating the ClusterCSIDriver is the correct way forward